tracer/src/Datadog.Trace/Ci/Coverage/Util/ArrayExtensions.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="ArrayExtensions.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Datadog.Trace.Ci.Coverage.Util;
+
+internal static class ArrayExtensions
+{
+#if NETCOREAPP3_0_OR_GREATER
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+#else
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+#endif
+    public static ref T FastGetReference<T>(this T[] array, int index)
```

Comment:
We should probably add a warning in a comment that this is unsafe, so it doesn't get used outside of safe locations.

Reply:
done in [edf6211](https://github.com/DataDog/dd-trace-dotnet/pull/3494/commits/edf62113ce850755976ceddb7999c68d56e0f4af)